### PR TITLE
Issue#2 - Save Image Link

### PR DESCRIPTION
**Pull request by Naman-sopho**
- Adds new dependency on `simple-form` gem
- A new page with a form to get user's image URL input
- Adds new model for `Image`

### CODE REVIEW

**Comment by Naman-sopho** [comment]
[Issue 2: Save Image Link](https://trello.com/c/rStuC2gy/409-issue-2-save-image-link)

File: app/controllers/image_controller.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class ImageController < ActionController::Base
+  def save; end
+end
```
Comment (bg43179) [inline app/controllers/image_controller.rb]: It seems we deleted some files and changed something in the next commit.
How about we do a `rebase` to merge this commit and the next commit? so we clean it up a little bit.

File: db/migrate/20210316202318_create_images.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class CreateImages < ActiveRecord::Migration[5.2]
+  def change
```
Comment (bg43179) [inline db/migrate/20210316202318_create_images.rb:2]: It's not necessary, but most of the time it's okay to put migration in its own commit.

File: Gemfile [hunk 3]
```diff
@@ -47,3 +47,5 @@ end
 
 # Windows does not include zoneinfo files, so bundle the tzinfo-data gem
 gem 'tzinfo-data', platforms: %i[mingw mswin x64_mingw jruby]
+
+gem 'simple_form'
```
Comment (bg43179) [inline Gemfile:51]: Let's move the `add simple_form gem` to its own commit, so we can separate dependency and production code change.

File: app/controllers/images_controller.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+class ImagesController < ActionController::Base
+  layout 'application'
+
+  def new
+    @image = Image.new
+  end
+
+  def create
+    @image = Image.new(link: params[:image][:link])
+    if @image.save
```
Comment (bg43179) [inline app/controllers/images_controller.rb:15]: This looks good!

Just want to share something you may be interested in: `save` vs `save!`
https://apidock.com/rails/v6.0.0/ActiveRecord/Persistence/save%21

File: test/fixtures/images.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (bg43179) [inline test/fixtures/images.yml:1]: Do we need this file?

File: app/controllers/images_controller.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+require 'simple_form'
```
Comment (bg43179) [inline app/controllers/images_controller.rb:1]: We may not need this dependency here

File: app/models/image.rb [hunk 7]
```diff
@@ -1,2 +1,4 @@
 class Image < ApplicationRecord
+  validates :link, presence: true
```
Comment (bg43179) [inline app/models/image.rb:2]: Since we have validation in the model, we will want to check if the validation actually works when saving images.

So some thoughts for tests we need:
1. Check if an image is valid or not (with link? without link)
2. If an image has a link, check if the link is a valid URL?

- ActiveModel has `valid?` method :https://apidock.com/rails/ActiveModel/Validations/valid%3F

**Review by hannaveldhuizen — APPROVED** [review]
